app/src/main/java/com/github/se/eduverse/repository/ProfileRepository.kt

```diff
@@ -214,6 +220,39 @@ open class ProfileRepositoryImpl(
     }
   }
 
+  override suspend fun loadSearchHistory(userId: String, limit: Int): List<Profile> {
+    return try {
+      val listId =
+          usersCollection
+              .document(userId)
+              .collection("searchHistory")
+              .orderBy("timestamp", Query.Direction.DESCENDING)
+              .limit(limit.toLong())
+              .get()
+              .await()
+              .documents
+              .mapNotNull { it.getString("id") }
+              .distinct()
+      profilesCollection
+          .whereIn(FieldPath.documentId(), listId)
+          .get()
+          .await()
+          .documents
+          .mapNotNull { it.toObject(Profile::class.java) }
+    } catch (e: Exception) {
+      Log.e("LOAD_HISTORY", "Failed to load search history: ${e.message}")
+      emptyList()
+    }
+  }
```

Comment:
The handling of exceptions in loadSearchHistory improves the robustness of the code by catching potential errors and logging them. well done